### DISCUSSION TITLE
Incorrect request_rate

### DISCUSSION THREAD

**Discussion by swainayasha** [artilleryio/artillery#2696] [Artillery]
Version info: 

VERSION INFO:

Artillery: 2.0.9
Node.js:   v18.20.2
OS:        darwin
Running this command: 
artillery run load-test-sgs.yaml
<command>

I expected to see this happen: Im expecting a request_rate of 5/sec with a total request of 600 and the script is running for 2min. but how is it for 3/sec in the summary report ??

I have the below arrival rate and ramp up time defined in my yaml script . Im expecting a request_rate of 5/sec but in the summary report the request_rate is 3/sec . How is that possible ? 
  phases:
    - name: Warm Up
      duration: 60
      arrivalRate: 5
    - name: Ramp Up
      duration: 60
      arrivalRate: 5
      rampTo: 5

All VUs finished. Total time: 2 minutes, 10 seconds

--------------------------------
Summary report @ 09:09:17(+0530)
--------------------------------

http.codes.200: ................................................................ 95
http.codes.500: ................................................................ 505
http.downloaded_bytes: ......................................................... 240055
http.request_rate: ............................................................. 3/sec
http.requests: ................................................................. 600
http.response_time:
  min: ......................................................................... 2327
  max: ......................................................................... 7342
  mean: ........................................................................ 6884.5
  median: ...................................................................... 7260.8
  p95: ......................................................................... 7260.8
  p99: ......................................................................... 7260.8
http.responses: ................................................................ 600
vusers.completed: .............................................................. 600
vusers.created: ................................................................ 600
vusers.created_by_name.Check Run Workflow for SearchCarExistornot: ............. 600
vusers.failed: ................................................................. 0
vusers.session_length:
  min: ......................................................................... 3085.9
  max: ......................................................................... 8379.3
  mean: ........................................................................ 7644
  median: ...................................................................... 8024.5
  p95: ......................................................................... 8024.5
  p99: ......................................................................... 8024.5

*explanation*

Instead, this happened: 

*explanation*

Files being used:
```
<relevant yaml/js/csv go here>
```

**Comment by hassy** [comment]
hi @swainayasha :wave: Arrival rate is not the same as the request rate. An arrival = a new VU. Each VU can make multiple requests. I would guess your script has VUs that make multiple requests. If you look at the `http.response_time` stats you'll see that the **fastest** request took 2.3s. So let's say VU make two requests one after another, and the first one takes 2.3s. That would give the request rate for one VU of: `2/2.3*` = `0.86` requests/second. Let's say you have 5 of those VUs arriving at the same time. You'd have 10 requests sent in 2.3s, so your request rate is `4.3` and not 5. In your specific case since requests can take up to 7.3s to be served, your overall request rate would be even lower.
